services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java

```diff
@@ -873,15 +873,15 @@ public static RouteOptions fromUrl(@NonNull URL url) {
 
     optionsJson.addProperty("baseUrl", url.getProtocol() + "://" + url.getHost());
 
-    String[] pathElements = url.getPath().split("/");
-    optionsJson.addProperty("user", pathElements[3]);
-    optionsJson.addProperty("profile", pathElements[4]);
-    optionsJson.addProperty("coordinates", pathElements[5]);
-
-    String[] queryElements = url.getQuery().split("&");
-    for (String query : queryElements) {
-      int idx = query.indexOf("=");
-      try {
+    try {
+      String[] pathElements = url.getPath().split("/");
+      optionsJson.addProperty("user", URLDecoder.decode(pathElements[3], UTF_8));
+      optionsJson.addProperty("profile", URLDecoder.decode(pathElements[4], UTF_8));
```

Comment:
Can `user` and `profile` contain semicolons?

Reply:
Not really, I just added it to keep things consistent :upside_down_face: